src/driver/xen.rs

```diff
@@ -49,6 +57,42 @@ impl Xen {
 
 impl api::Introspectable for Xen {
 
+    fn read_physical(&self, paddr: u64, count: u32) -> Result<Vec<u8>,&str> {
```

Comment:
This implementation comes from libvmi:
https://github.com/libvmi/libvmi/blob/master/libvmi/read.c#L116